File: Assets/Code/Outfit7/Raven/TrackComponents/Properties/Base/RavenAnimationPropertyComponentBase.cc
#ifdef STARLITE
#include "RavenAnimationPropertyComponentBase.h"
#include "RavenAnimationPropertyComponentBase.cs"

#include <RavenSequence.h>

namespace Starlite {
    namespace Raven {
        RavenAnimationPropertyComponentBase::RavenAnimationPropertyComponentBase() {
        }

        bool RavenAnimationPropertyComponentBase::IsTriggerPropertyValid() const {
            return m_TriggerProperty == nullptr || m_TriggerProperty->GetTargetHash() != RavenUtility::s_InvalidHash;
        }

        bool RavenAnimationPropertyComponentBase::IsValid() const {
            bool propertyValid = m_AnimationData != nullptr && (IsCustom() ? IsCustomValid() : m_TargetHash != RavenUtility::s_InvalidHash);
            return propertyValid && IsTriggerPropertyValid();
        }

        const RavenPropertyComponent* RavenAnimationPropertyComponentBase::GetChildPropertyComponent() const {
            return m_TriggerProperty;
        }

        String RavenAnimationPropertyComponentBase::GetComponentType() const {
            return m_ComponentType;
        }

        String RavenAnimationPropertyComponentBase::GetMemberName() {
            return m_MemberName;
        }

        const String& RavenAnimationPropertyComponentBase::GetMemberName() const {
            return m_MemberName;
        }

        UInt64 RavenAnimationPropertyComponentBase::GetTargetHash() const {
            return m_TargetHash;
        }

        void RavenAnimationPropertyComponentBase::DestroyEditor(Ptr<RavenSequence> sequence) {
        }

        void RavenAnimationPropertyComponentBase::Initialize(Ptr<RavenSequence> sequence) {
            RavenPropertyComponent::Initialize(sequence);
            if (m_TriggerProperty != nullptr) {
                m_TriggerProperty->Initialize(sequence);
            }
            if (m_AnimationData != nullptr) {
                m_AnimationData->Initialize(sequence, this);
            }
        }

        void RavenAnimationPropertyComponentBase::SetComponentType(String value) {
            m_ComponentType = value;
        }

        void RavenAnimationPropertyComponentBase::SetMemberName(String value) {
            m_MemberName = value;
        }

        void RavenAnimationPropertyComponentBase::SetTargetHash(UInt64 value) {
            m_TargetHash = value;
        }
    } // namespace Raven
} // namespace Starlite
#endif

File: Assets/Code/Outfit7/Raven/TrackComponents/Properties/Base/RavenAnimationPropertyComponentBase.cs
//{ "engine_register_objects":["Starlite__Raven__RavenAnimationPropertyComponentBase"] }
#if !__cplusplus
#pragma warning disable
namespace Starlite {

    namespace Raven {
    
        // RavenAnimationPropertyComponentBase
        [Starlite.ObjectClassAttribute("Starlite::Raven::RavenAnimationPropertyComponentBase")]
        public abstract partial class RavenAnimationPropertyComponentBase : global::Starlite.Raven.RavenPropertyComponent {
        	// Properties
        	[UnityEngine.SerializeField] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] protected global::Starlite.Raven.RavenTriggerPropertyComponentBase m_TriggerProperty;
        	[UnityEngine.SerializeField] [UnityEngine.HeaderAttribute("Internal")] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] protected global::Starlite.Raven.RavenAnimationDataComponentBase m_AnimationData;
        	[UnityEngine.SerializeField] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] private string m_ComponentType;
        	[UnityEngine.SerializeField] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] private string m_MemberName;
        	[UnityEngine.SerializeField] [Starlite.ObjectPropertyAttribute(Starlite.SourceCCProperty.PropertyAttribute.Preload)] private ulong m_TargetHash;
        }
        
    }
    
}
#pragma warning restore
#else
// RavenAnimationPropertyComponentBase
namespace Starlite { namespace Raven { 
SIMPLEMENT_CLASS_ABSTRACT_WITH_PROPERTIES_IN_CLASS_SCOPED(Starlite::Raven::RavenAnimationPropertyComponentBase, RavenAnimationPropertyComponentBase, RavenAnimationPropertyComponentBase, Starlite::Raven::RavenPropertyComponent, Starlite::SceneObjectComponent, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationPropertyComponentBase, m_TriggerProperty, m_TriggerProperty, kObject, kUnknown, Starlite::Raven::RavenTriggerPropertyComponentBase::TypeId, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationPropertyComponentBase, m_AnimationData, m_AnimationData, kObject, kUnknown, Starlite::Raven::RavenAnimationDataComponentBase::TypeId, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationPropertyComponentBase, m_ComponentType, m_ComponentType, kString, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationPropertyComponentBase, m_MemberName, m_MemberName, kString, kUnknown, -1, "")
SIMPLEMENT_PROPERTY_UNIVERSAL(RavenAnimationPropertyComponentBase, m_TargetHash, m_TargetHash, kInt64, kUnknown, -1, "")
SIMPLEMENT_CLASS_END(RavenAnimationPropertyComponentBase)
} } 
SIMPLEMENT_CLASS_SCOPE(Starlite__Raven__RavenAnimationPropertyComponentBase, Starlite::Raven, RavenAnimationPropertyComponentBase)
#endif
